sys/net/routing/nhdp/nhdp.c

```diff
@@ -35,6 +36,10 @@
 #include "nhdp_writer.h"
 #include "nhdp_reader.h"
 
+#ifndef MODULE_CONN_UDP
+#error "nhdp needs a conn_udp implementation to work"
```

Comment:
That would defeat conn's purpose of being an interface agnostic to a stack implementation (similar to sockets just light weight). The pseudo-module `conn_udp` is pulled in by any module called `%_conn_udp`, with `%` being a wildcard. If this wildcard is `gnrc` or `openwsn` or `lwip` or @kaspar030's infamous nano-stack doesn't matter, since that is exactly the point of that interface. Since `conn_udp` however doesn't supply any implementation on itself a user trying to compile this module without any conn interface would get more or less cryptic error messages and assume it is broken. So a friendly error message seemed to me to be the better solution :)